### PR TITLE
[Calling][Feature] Rooms integration phase 01

### PR DESCRIPTION
**Pull request by Mohtasim**
## Purpose

* Adds capability to join a rooms call as an attendee

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull Request Checklist

- [x] Public API changes
- [ ] Verified for cross-platform
- [ ] Synced with iOS, Web team
- [ ] Internal review completed
- [ ] UI Changes
  - [ ] Screen captures included
  - [ ] Tested for Light/Dark mode
  - [ ] Tested for screen rotation
  - [ ] Tested on Tablet and small screen device (5")
  - [ ] Include localization changes
- [ ] Tests
  - [ ] Memory leak analysis performed
  - [ ] Tested on API 21, 26 and latest 
  - [ ] Tested for background mode
  - [ ] Unit Tests Included
  - [ ] UI Automated Tests Included

## How to Test

* Open demo app
* enter the room number and acs token
* join rooms call

## What to Check
Verify that the following are valid
* If user can join an interact in a rooms call

### CODE REVIEW

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/setup/components/ErrorInfoView.kt [hunk 1]
```diff
@@ -76,9 +76,6 @@ internal class ErrorInfoView(private val rootView: View) {
         if (errorMessage.isBlank()) return
         snackBarTextView.text = errorMessage
         snackBar.run {
-            if (isShown) {
```
Comment (Mohtasim) [inline azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/setup/components/ErrorInfoView.kt:79]: Why was this removed?

Comment (pavelprystinka) [reply azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/setup/components/ErrorInfoView.kt:79]: If snack bar is already shown, then it does close it, but a new one is not opened. 
From user perspective we do not need to close and open, the only what needs to change is a message.

**Review by pavelprystinka — APPROVED** [review]

**Review by ShaunaSong — APPROVED** [review]
